config/initializers/rpush.rb

```diff
@@ -69,6 +77,8 @@
       tags: [
         "error:#{e.class}",
         "message:#{e.error_description}",
+        "app_id:#{notification.app_id}",
+        "bundle_id:#{notification&.app&.bundle_id}",
```

Comment:
Based on the comment I left above (with the data we can access in these callbacks) I think we can change this to something along the lines of:

```ruby
"error:#{notification.error_code}",
"message:#{notification.error_description}",
"app_bundle:#{notification.app&.bundle_id}",
```

The reason I'm sugggesting `app_bundle` instead of `bundle_id` is because that's the name we have in the  Device model (`app_bundle`). Also, `app_id ` is an integer identifier within Redis and it doesn't give us much information (it will be an arbitrary `1`, `8`, `17`, etc), so the real visibility would come from `app_bundle`.

Let me know what you think 🙂

Reply:
Awesome, thanks so much for this info! Finding specifics on what exactly is available on the `Notification` object has been really hard: RPush doesn't really document it very well, and the code is pretty unclear.